selenium_test.py

```diff
@@ -0,0 +1,222 @@
+        enabledButton = driver.find_elements(By.XPATH, '//*[@title="Next"]')
+        print("Length of enabled button" + str(len(enabledButton)))
+        print("Length of diabled button" + str(len(disabledButton)))
+
+        if len(enabledButton) > 0:
+            enabledButton[0].click()
+            collectResults(driver, keyword)
+            hasNextPage = True
+        elif len(disabledButton) > 0:
+            collectResults(driver, keyword)
+            break
+
+    # waits for the new rollover download
+    markedWait = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.CLASS_NAME, "markListSpan"))
+    )
+
+    # clicks marked items button
+    markedItems = driver.find_element_by_class_name("markListSpan")
+    markedItems.click()
+
+    # clicks marked items button
+    download = driver.find_element_by_id("download")
+    download.click()
+
+    currentWindow = driver.current_window_handle
+    downloadPage = driver.window_handles[1]
+    driver.switch_to.window(downloadPage)
+
+    # waits for the new rollover download
+    resultsWait = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.CLASS_NAME, "rolloverdownload"))
+    )
+
+    # clicks marked items button
+    rolloverdownload = driver.find_element_by_class_name("rolloverdownload")
+    rolloverdownload.click()
+
+    driver.switch_to.window(currentWindow)
+
+    df = pd.DataFrame(list(zip(lists.title_list, lists.description_list, lists.category_list, lists.keyword_list, lists.link_list)),
+                      columns=['Title', 'Description', 'Category', 'Keyword', "link"])
+
+    df.to_csv('selenium-output.csv', index=True)
+
+
+
+
+
+
+
+
+
+
+def collectResults(driver, keyword):
+    # waits for the presence of the Ul item  on new results page
+    time.sleep(1)
+    resultsWait = WebDriverWait(driver, 30).until(
+        EC.presence_of_element_located((By.CLASS_NAME, "resultsListBox"))
+    )
+
+    # grabs list item search results
+    resultItems = driver.find_elements_by_class_name("resultsListBox")
+
+
+    for result in resultItems:
+        title = result.find_element_by_class_name("articleTitle").text
+        description = result.find_element_by_class_name("description").text
+        category = result.find_element_by_class_name("articleType").text
+        link = result.find_element_by_class_name("articleTitle").find_element_by_css_selector('a').get_attribute('href')
+
+
+
+
+
+
+
+
+        lists.link_list.append(link)
+        lists.title_list.append(title)
+        lists.description_list.append(description)
+        lists.category_list.append(category)
+        lists.keyword_list.append(keyword)
+
+
+        markItem = result.find_element_by_class_name("markItem")
+        markItem.click()
+
+    return lists
+
+
+
+
+
+
+def groupResults():
+    df = pd.read_csv('selenium-output.csv', usecols=['Title','Description','Category','Keyword',"link"])
```

Comment:
I don't understand why you write the outputs to a temporary csv, then read them back in. This was necessary for Scrapy because Scrapy can only output to a file, but in this case data should be passed in through arguments or returned from functions.